addons/hw_drivers/iot_handlers/drivers/PrinterDriver_L.py

```diff
@@ -399,7 +399,10 @@ def open_cashbox(self, data):
 
     def _action_default(self, data):
         self.print_raw(b64decode(data['document']))
-        send_to_controller(self.connection_type, {'print_id': data['print_id'], 'device_identifier': self.device_identifier})
+        WebsocketClient.send_to_controller({
+            'operation_id': data['operation_id'],
+            'device_identifier': self.device_identifier
+        })
```

Comment:
I do wonder if it would not be better to keep the `send_to_controller` logic in the core code (websocket.py rather than drivers). I don't really understand the logic as of why the drivers should trigger the confirmation of the request;
first it could be called by the HTTP route, then the call to the server is useless as it was not expecting a return through websocket (to my knowledge).
Second, it would make more sense to try: catch: the IoT operation in websocket.py and send something back to the server (wherever the operation was successful or not with error message details if any).

I could understand the logic if the functions were adding parameters only known to the drivers, but it does not look to be the case

Reply:
Done

Reply:
@lse-odoo Just added the try-catch logic, I still need to test (will do it next week) but I don't really like the `except: Exception`...